app/process/watcher/crypto-transfer/crypto-transfer.go

```diff
@@ -94,109 +99,112 @@ func (ctw CryptoTransferWatcher) getTimestamp(q *queue.Queue) int64 {
 			ctw.restart(q)
 			return
 		}
 
 		if len(transactions.Transactions) > 0 {
 			for _, tx := range transactions.Transactions {
-				log.Infof("[%s] - New transaction on account [%s] - Tx Hash: [%s]",
-					tx.ConsensusTimestamp,
-					ctw.accountID.String(),
-					tx.TransactionHash)
-
-				var amount int64
-				for _, tr := range tx.Transfers {
-					if tr.Account == ctw.accountID.String() {
-						amount = tr.Amount
-					}
-				}
-
-				decodedMemo, e := base64.StdEncoding.DecodeString(tx.MemoBase64)
-				if e != nil || len(decodedMemo) < 42 {
-					log.Errorf("[%s] Crypto Transfer Watcher: Could not verify transaction memo - Error: [%s]\n", ctw.accountID.String(), e)
-					continue
-				}
-
-				// TODO: Should verify memo.
-				ethAddress := decodedMemo[:42]
-				feeString := string(decodedMemo[42:])
-
-				re := regexp.MustCompile("^0x[0-9a-fA-F]{40}$")
-				if !re.MatchString(string(ethAddress)) {
-					log.Errorf("[%s] Crypto Transfer Watcher: Could not verify Ethereum Address\n\t- [%s]", ctw.accountID.String(), ethAddress)
-					continue
-				}
-
-				_, e = helper.ToBigInt(feeString)
-				if e != nil {
-					log.Errorf("[%s] Crypto Transfer Watcher: Could not verify transaction fee\n\t- [%s]", ctw.accountID.String(), feeString)
-					continue
-				}
-
-				stateProof, e := ctw.client.GetStateProof(tx.TransactionID)
-				if e != nil {
-					log.Errorf("[%s] Crypto Transfer Watcher: Could not GET state proof, TransactionID [%s]. Error [%s]", ctw.accountID.String(), tx.TransactionID, e)
-					continue
-				}
-
-				verified, e := stateproof.Verify(tx.TransactionID, stateProof)
-				if e != nil {
-					log.Errorf("[%s] Crypto Transfer Watcher: Error while trying to verify state proof for TransactionID [%s]. Error [%s]", ctw.accountID.String(), tx.TransactionID, e)
-					continue
-				}
-
-				if !verified {
-					log.Errorf("[%s] Crypto Transfer Watcher: Failed to verify state proof for TransactionID [%s]", ctw.accountID.String(), tx.TransactionID)
-					continue
-				}
-
-				information := &protomsg.CryptoTransferMessage{
-					TransactionId: tx.TransactionID,
-					EthAddress:    string(ethAddress),
-					Amount:        uint64(amount),
-					Fee:           feeString,
-				}
-				publisher.Publish(information, ctw.typeMessage, ctw.accountID, q)
+				go ctw.processTransaction(tx, q)
 			}
 			var err error
 			milestoneTimestamp, err = timestamp.FromString(transactions.Transactions[len(transactions.Transactions)-1].ConsensusTimestamp)
 			if err != nil {
-				log.Errorf("[%s] Crypto Transfer Watcher: [%s]", ctw.accountID.String(), err)
+				ctw.logger.Errorf("[%s]", ctw.accountID.String(), err)
 				continue
 			}
 		}
 
 		err := ctw.statusRepository.UpdateLastFetchedTimestamp(ctw.accountID.String(), milestoneTimestamp)
 		if err != nil {
-			log.Errorf("Error incoming: Suddenly stopped monitoring account [%s] - [%s]", ctw.accountID.String(), e)
+			ctw.logger.Errorf("Error incoming: Suddenly stopped monitoring account - [%s]", e)
```

Comment:
Error message should not be regarding monitoring, but saying that has failed to update latest fetched timestamp.